dbms/src/Storages/DeltaMerge/FilterParser/FilterParser.cpp

```diff
@@ -93,6 +93,19 @@ ColumnID getColumnIDForColumnExpr(const tipb::Expr & expr, const ColumnDefines &
     return columns_to_read[column_index].id;
 }
 
+ColumnDefine getColumnDefineForColumnExpr(const tipb::Expr & expr, const ColumnDefines & columns_to_read)
+{
+    assert(isColumnExpr(expr));
+    auto column_index = decodeDAGInt64(expr.val());
+    if (column_index < 0 || column_index >= static_cast<Int64>(columns_to_read.size()))
+    {
+        throw TiFlashException("Column index out of bound: " + DB::toString(column_index) + ", should in [0,"
+                                   + DB::toString(columns_to_read.size()) + ")",
+                               Errors::Coprocessor::BadRequest);
+    }
+    return columns_to_read[column_index];
+}
```

Comment:
unnecessrary, just call `auto column_index = decodeDAGInt64(expr.val());` to get column index.

Reply:
It is more concise to encapsulate it into a function.